src/typing_manager.ts

```diff
@@ -192,8 +207,10 @@ export class TypingManager implements Disposable {
     private onEscapeKeyCommand = async (key = "<Esc>"): Promise<void> => {
         // rebind early to store fast pressed keys which may happen between sending changes to neovim and exiting insert mode
         // see https://github.com/asvetliakov/vscode-neovim/issues/324
-        this.isExitingInsertMode = true;
-        await this.onSendBlockingCommand(key);
+        if (this.main.modeManager.neovimToggle) {
```

Comment:
This will break `<C-o>` in insert mode from working. Id say this would be fine if this PR totally disables neovim, but currently it just keeps nvim stuck in insert mode, meaning other commands like `<C-r>`/`<C-w>`/etc still work.

Probably just do `|| key !== "<Esc>"`.

Reply:
tried with both setting "Vscode-neovim: Use Ctrl Keys For Insert Mode" on and off.. the keys C-r, C-w and C-o do what is stated in the Keyboard shortcut stated. It only enters this function when pressing <Esc> or when i do jk which is the same. maybe it changes after restart Vscode completly.. I just reloaded the window to let the  "Vscode-neovim: Use Ctrl Keys For Insert Mode" Setting going active

Reply:
- settings only apply upon vscode restart
- all the keys do work normally when `toggle` is true
- C-r/C-r works normally when `toggle` is false
- C-o does NOT work when `toggle` is false

Reply:
okay, i could reproduce after i disabled karabiner on mac.
It's as you said. Fixed now